gui-agent/vmside.c

```diff
@@ -1976,13 +1979,32 @@ static pid_t do_execute_xorg(
     }
 }
 
-static void terminate_and_cleanup_xorg(Ghandles *g) {
+/* this cannot be a SIGCHLD signal handler because of signal safety */
+static void *xorg_waitpid_thread(void *p) {
+    Ghandles *g = p;
     int status;
+    waitpid(g->x_pid, &status, 0);
```

Comment:
What if this fails?

Reply:
Added some error handling here. Not sure if it's the *best* strategy, but at least we're not failing silently.